### PR TITLE
Added support for Slice Caffe function.

### PR DESCRIPTION
**Pull request by amitibo**
Add Caffe ``Slice`` layer

### CODE REVIEW

**Comment by rezoo** [comment]
Thank you for your PR! I will check it.

File: chainer/links/caffe/caffe_function.py [hunk 1]
```diff
@@ -384,6 +384,28 @@ def _setup_scale(self, layer):
...
+            axis = layer.slice_param.axis
+        elif layer.slice_param.HasField('slice_dim'):
+            axis = layer.slice_param.slice_dim
+        else:
+            axis = 1
+
+        if layer.slice_param.slice_point == []:
+            indices_or_sections = len(list(layer.top))
+        else:
+            indices_or_sections = list(layer.slice_params.slice_point)
```
Comment (rezoo) [inline chainer/links/caffe/caffe_function.py]: Could you replace ``layer.slice_params.slice_point`` with ``layer.slice_param.slice_point``?

File: tests/chainer_tests/links_tests/caffe_tests/test_caffe_function.py [hunk 2]
```diff
@@ -920,6 +920,37 @@ def test_scale_one_bottom_with_bias(self):
...
+class TestSlice(TestCaffeFunctionBaseMock):
+
+    func_name = 'chainer.functions.split_axis'
+    in_shapes = [(3, 4, 3)]
+    out_shapes = [(3, 2, 3), (3, 2, 3)]
+
+    data = {
+        'layer': [
+            {
+                'name': 'slice',
```
Comment (rezoo) [inline tests/chainer_tests/links_tests/caffe_tests/test_caffe_function.py]: In this case the layer name``slice`` should be renamed ``l1`` like the other layers.

**Comment by rezoo** [comment]
LGTM except for the above comments!

File: chainer/links/caffe/caffe_function.py [hunk 3]
```diff
@@ -384,6 +384,28 @@ def _setup_scale(self, layer):
...
+    @_layer('Slice', 'SLICE')
+    def _setup_slice(self, layer):
+        if layer.slice_param.HasField('axis'):
+            axis = layer.slice_param.axis
+        elif layer.slice_param.HasField('slice_dim'):
+            axis = layer.slice_param.slice_dim
+        else:
+            axis = 1
+
+        if layer.slice_param.slice_point == []:
```
Comment (delta2323) [inline chainer/links/caffe/caffe_function.py]: `if layer.slice_param.slice_point:` is simpler to check the list is empty.

**Comment by delta2323** [comment]
Please add the following test cases:
* `axis` does not exist and `slice_dim` exist.
* Both `axis` nor `slice_dim` do not exist.
* `slice_point` is not empty.

**Comment by delta2323** [comment]
I added comments. Could you check them, please?

**Comment by rezoo** [comment]
Thank you so much! I would like to merge your PR 👍

**Pull request merged**